zerver/tests/test_zulip_update_announcements.py

```diff
@@ -202,7 +223,10 @@ def test_group_direct_message_with_zulip_updates_stream_set(self) -> None:
         expected_group_direct_message_user_ids = admin_user_ids | {notification_bot.id}
 
         now = timezone_now()
-        with time_machine.travel(now, tick=False):
+        with time_machine.travel(now, tick=False), mock.patch(
+            "zerver.lib.zulip_update_announcements.zulip_update_announcements",
+            self.zulip_update_announcements,
+        ):
```

Comment:
This pattern might be cleaner just using `@mock.patch` on the individual functions, if we want the scope of the changes to be for the entire test?

Reply:
`self.zulip_update_announcements` instance variable is not accessible in decorators -- hence used this approach.

Couple of other approaches I have in my mind:
* All the test code within a `with mock.patch()` context manager at the start of the test.
* Current approach: See `def setUp()`

Reply:
Ahh, OK. This is probably still a bit more robust.